packages/agents/src/HfAgent.ts

```diff
@@ -0,0 +1,87 @@
+import { HfInference } from "../../inference/src";
+import type { TextGenerationOutput } from "../../inference/src";
+
+import { evalBuilder } from "./lib/evalBuilder";
+import { generateCode } from "./lib/generateCode";
+import { defaultTools } from "./tools";
+import type { Tool, Update } from "./types";
+
+export interface LLMFromHub {
+	model: string;
+}
+export interface LLMFromEndpoint {
+	endpoint: string;
+}
+export type LLMSettings = LLMFromEndpoint | LLMFromHub;
+
+function isLLMFromHub(settings: LLMSettings): settings is LLMFromHub {
+	return (settings as LLMFromHub).model !== undefined;
+}
+
+export class HfAgent {
+	private accessToken: string;
+	private settings: LLMSettings;
+	private tools: Tool[];
+
+	constructor(accessToken = "", settings?: LLMSettings, tools?: Tool[]) {
+		this.accessToken = accessToken;
+		this.settings = settings ?? { model: "OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5" };
+		this.tools = tools ?? defaultTools;
+	}
+
+	public async LLM(prompt: string): Promise<string> {
+		const formattedPrompt = "<|user|>" + prompt + "<|end|><|assistant|>";
```

Comment:
would need changes to make it compatible with Llama2 etc? (can be in a later PR)

Reply:
Refactored things so it would be easy to add support for llama2 in a laterPR!